src/attended-event/attended-event.service.spec.ts

```diff
@@ -0,0 +1,187 @@
+  async findByAuthId() {
+    /* Should be overriden using spyOn */
+  }
+  async update() {
+    /* Should be overriden using spyOn */
+  }
+}
+
+class MockEventService {
+  async create() {
+    /* Should be overriden using spyOn */
+  }
+  async findAll() {
+    /* Should be overriden using spyOn */
+  }
+  async findById() {
+    /* Should be overriden using spyOn */
+  }
+  async update() {
+    /* Should be overriden using spyOn */
+  }
+}
+
+describe("AttendedEventService", () => {
+  let service: AttendedEventService;
+  let userService: UserService;
+  let eventService: EventService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        TestDatabaseModule,
+        MongooseModule.forFeature([
+          { name: "AttendedEvent", schema: AttendedEventSchema }
+        ])
+      ],
+      providers: [
+        AttendedEventService,
+        { provide: UserService, useValue: new MockUserService() },
+        { provide: EventService, useValue: new MockEventService() }
+      ]
+    }).compile();
+
+    service = module.get<AttendedEventService>(AttendedEventService);
+    userService = module.get<UserService>(UserService);
+    eventService = module.get<EventService>(EventService);
+  });
+
+  it("should be defined", () => {
+    expect(service).toBeDefined();
+    expect(userService).toBeDefined();
+    expect(eventService).toBeDefined();
+  });
+
+  it("should record an AttendedEvent given a valid userAuthId and eventId", async () => {
+    jest.spyOn(userService, "findByAuthId").mockImplementation(
+      async (): Promise<User> =>
+        Promise.resolve({
+          authId: "random",
+          email: "example@example.com",
+          notificationEmail: "example@example.com"
+        } as User)
+    );
+    jest.spyOn(eventService, "findById").mockImplementation(
+      async (): Promise<Event> =>
+        Promise.resolve({
+          name: "A Random Event",
+          parentId: "root",
+          description: "",
+          id: "randomEventId"
+        } as Event)
+    );
+
+    const attendedEvent = await service.record({
+      eventId: "randomEventId",
+      authId: "random"
+    });
+
+    expect(attendedEvent).toBeDefined();
+    expect(attendedEvent.eventId).toBe("randomEventId");
+    expect(attendedEvent.userAuthId).toBe("random");
+  });
+
+  it("should fail to record an AttendedEvent if user doesn't exist", async () => {
+    jest
+      .spyOn(userService, "findByAuthId")
+      .mockImplementation(async (): Promise<User | undefined> => undefined);
+    jest.spyOn(eventService, "findById").mockImplementation(
+      async (): Promise<Event> =>
+        Promise.resolve({
+          name: "A Random Event",
+          parentId: "root",
+          description: "",
+          id: "randomEventId"
+        } as Event)
+    );
+
+    try {
+      await service.record({
+        eventId: "randomEventId",
+        authId: "random"
+      });
+    } catch (e) {
```

Comment:
An issue with testing for errors this way is that if the `try` block doesn't throw an error, then the `catch` block doesn't run and that test passes when it shouldn't. Using the `toThrow` jest function might be better here.

Check out [this](https://github.com/tamu-datathon-org/gatekeeper/blob/b53a4e0a541333dae9c8e7e43385bfdcbfe00c29/src/user-auth/user-auth.service.spec.ts#L66) for an example in `user-auth.service.spec.ts`